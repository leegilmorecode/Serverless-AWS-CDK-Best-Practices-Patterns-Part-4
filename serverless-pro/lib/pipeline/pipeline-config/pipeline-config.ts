import * as dotenv from 'dotenv';

import { Region, Stage } from '../../types';

import { EnvironmentConfig } from '../pipeline-types/pipeline-types';

dotenv.config();

export const environments: Record<Stage, EnvironmentConfig> = {
  // allow developers to spin up a quick branch for a given PR they are working on e.g. pr-124
  // this is done with a npm run develop, not through the pipeline, and uses the values in .env
  [Stage.develop]: {
    env: {
      account:
        process.env.ACCOUNT || (process.env.CDK_DEFAULT_ACCOUNT as string),
      region: process.env.REGION || (process.env.CDK_DEFAULT_REGION as string),
    },
    stateful: {
      bucketName:
        `serverless-pro-${process.env.PR_NUMBER}-bucket`.toLowerCase(),
      assetsBucketName:
        `serverless-pro-${process.env.PR_NUMBER}-canary-bucket`.toLowerCase(),
    },
    stateless: {
      lambdaMemorySize: parseInt(process.env.LAMBDA_MEM_SIZE || '128'),
      canaryNotificationEmail: process.env.NOTIFICATION_EMAIL as string,
      randomErrorsEnabled: process.env.RANDOM_ERRORS_ENABLED || 'false',
    },
    client: {
      bucketName:
        `serverless-pro-client-${process.env.PR_NUMBER}-bucket`.toLowerCase(),
    },
    shared: {
      domainName: process.env.DOMAIN_NAME as string,
      domainCertificateArn: process.env.DOMAIN_CERT_ARN as string,
      appConfigLambdaLayerArn: process.env
        .APP_CONFIG_LAMBDA_LAYER_ARN as string,
      powerToolsMetricsNamespace: process.env
        .POWERTOOLS_METRICS_NAMESPACE as string,
      powerToolServiceName: process.env.POWERTOOLS_SERVICE_NAME as string,
    },
    stageName: process.env.PR_NUMBER || Stage.develop,
  },
  [Stage.featureDev]: {
    env: {
      account: '123456789123',
      region: Region.dublin,
    },
    stateful: {
      bucketName: 'serverless-pro-feature-dev-bucket',
      assetsBucketName:
        `serverless-pro-feature-dev-canary-bucket`.toLowerCase(),
    },
    stateless: {
      lambdaMemorySize: 128,
      canaryNotificationEmail: 'your-email@gmail.com',
      randomErrorsEnabled: 'false',
    },
    client: {
      bucketName: 'serverless-pro-client-feature-dev-bucket',
    },
    shared: {
      domainName: 'your-domain.co.uk',
      domainCertificateArn:
        'arn:aws:acm:us-east-1:123456789123:certificate/3c0a6045-j786-iu87-rt45-74e87b1e6099',
      appConfigLambdaLayerArn:
        'arn:aws:lambda:eu-west-1:434848589818:layer:AWS-AppConfig-Extension-Arm64:46',
      powerToolServiceName: 'serverless-pro-orders-service-feature-dev',
      powerToolsMetricsNamespace: 'ServerlessProFeatureDev',
    },
    stageName: Stage.featureDev,
  },
  [Stage.staging]: {
    env: {
      account: '123456789123',
      region: Region.dublin,
    },
    stateful: {
      bucketName: 'serverless-pro-staging-bucket',
      assetsBucketName: `serverless-pro-staging-canary-bucket`.toLowerCase(),
    },
    stateless: {
      lambdaMemorySize: 1024,
      canaryNotificationEmail: 'your-email@gmail.com',
      randomErrorsEnabled: 'false',
    },
    client: {
      bucketName: 'serverless-pro-client-staging-bucket',
    },
    shared: {
      domainName: 'your-domain.co.uk',
      domainCertificateArn:
        'arn:aws:acm:us-east-1:123456789123:certificate/3c0a6045-j786-iu87-rt45-74e87b1e6099',
      appConfigLambdaLayerArn:
        'arn:aws:lambda:eu-west-1:434848589818:layer:AWS-AppConfig-Extension-Arm64:46',
      powerToolServiceName: 'serverless-pro-orders-service-staging',
      powerToolsMetricsNamespace: 'ServerlessProStaging',
    },
    stageName: Stage.staging,
  },
  [Stage.prod]: {
    env: {
      account: '123456789123',
      region: Region.dublin,
    },
    stateful: {
      bucketName: 'serverless-pro-prod-bucket',
      assetsBucketName: `serverless-pro-prod-canary-bucket`.toLowerCase(),
    },
    stateless: {
      lambdaMemorySize: 1024,
      canaryNotificationEmail: 'your-email@gmail.com',
      randomErrorsEnabled: 'false',
    },
    client: {
      bucketName: 'serverless-pro-client-prod-bucket',
    },
    shared: {
      domainName: 'your-domain.co.uk',
      domainCertificateArn:
        'arn:aws:acm:us-east-1:123456789123:certificate/3c0a6045-j786-iu87-rt45-74e87b1e6099',
      appConfigLambdaLayerArn:
        'arn:aws:lambda:eu-west-1:434848589818:layer:AWS-AppConfig-Extension-Arm64:46',
      powerToolServiceName: 'serverless-pro-orders-service-prod',
      powerToolsMetricsNamespace: 'ServerlessProProd',
    },
    stageName: Stage.prod,
  },
};
