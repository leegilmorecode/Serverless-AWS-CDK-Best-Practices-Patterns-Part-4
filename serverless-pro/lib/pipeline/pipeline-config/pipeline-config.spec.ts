// https://github.com/cypress-io/cypress/issues/22059
import { describe, expect, it } from '@jest/globals';

import { environments } from './pipeline-config';

describe('pipeline-config', () => {
  it('should return the correct config for feature-dev', () => {
    expect(environments.featureDev).toMatchSnapshot();
  });

  it('should return the correct config for staging', () => {
    expect(environments.staging).toMatchSnapshot();
  });

  it('should return the correct config for prod', () => {
    expect(environments.prod).toMatchSnapshot();
  });
});
