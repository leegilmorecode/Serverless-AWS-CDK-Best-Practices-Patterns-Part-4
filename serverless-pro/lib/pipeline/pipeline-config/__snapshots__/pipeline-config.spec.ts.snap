// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pipeline-config should return the correct config for feature-dev 1`] = `
{
  "client": {
    "bucketName": "serverless-pro-client-feature-dev-bucket",
  },
  "env": {
    "account": "123456789123",
    "region": "eu-west-1",
  },
  "shared": {
    "appConfigLambdaLayerArn": "arn:aws:lambda:eu-west-1:434848589818:layer:AWS-AppConfig-Extension-Arm64:46",
    "domainCertificateArn": "arn:aws:acm:us-east-1:123456789123:certificate/3c0a6045-j786-iu87-rt45-74e87b1e6099",
    "domainName": "your-domain.co.uk",
    "powerToolServiceName": "serverless-pro-orders-service-feature-dev",
    "powerToolsMetricsNamespace": "ServerlessProFeatureDev",
  },
  "stageName": "featureDev",
  "stateful": {
    "assetsBucketName": "serverless-pro-feature-dev-canary-bucket",
    "bucketName": "serverless-pro-feature-dev-bucket",
  },
  "stateless": {
    "canaryNotificationEmail": "your-email@gmail.com",
    "lambdaMemorySize": 128,
    "randomErrorsEnabled": "false",
  },
}
`;

exports[`pipeline-config should return the correct config for prod 1`] = `
{
  "client": {
    "bucketName": "serverless-pro-client-prod-bucket",
  },
  "env": {
    "account": "123456789123",
    "region": "eu-west-1",
  },
  "shared": {
    "appConfigLambdaLayerArn": "arn:aws:lambda:eu-west-1:434848589818:layer:AWS-AppConfig-Extension-Arm64:46",
    "domainCertificateArn": "arn:aws:acm:us-east-1:123456789123:certificate/3c0a6045-j786-iu87-rt45-74e87b1e6099",
    "domainName": "your-domain.co.uk",
    "powerToolServiceName": "serverless-pro-orders-service-prod",
    "powerToolsMetricsNamespace": "ServerlessProProd",
  },
  "stageName": "prod",
  "stateful": {
    "assetsBucketName": "serverless-pro-prod-canary-bucket",
    "bucketName": "serverless-pro-prod-bucket",
  },
  "stateless": {
    "canaryNotificationEmail": "your-email@gmail.com",
    "lambdaMemorySize": 1024,
    "randomErrorsEnabled": "false",
  },
}
`;

exports[`pipeline-config should return the correct config for staging 1`] = `
{
  "client": {
    "bucketName": "serverless-pro-client-staging-bucket",
  },
  "env": {
    "account": "123456789123",
    "region": "eu-west-1",
  },
  "shared": {
    "appConfigLambdaLayerArn": "arn:aws:lambda:eu-west-1:434848589818:layer:AWS-AppConfig-Extension-Arm64:46",
    "domainCertificateArn": "arn:aws:acm:us-east-1:123456789123:certificate/3c0a6045-j786-iu87-rt45-74e87b1e6099",
    "domainName": "your-domain.co.uk",
    "powerToolServiceName": "serverless-pro-orders-service-staging",
    "powerToolsMetricsNamespace": "ServerlessProStaging",
  },
  "stageName": "staging",
  "stateful": {
    "assetsBucketName": "serverless-pro-staging-canary-bucket",
    "bucketName": "serverless-pro-staging-bucket",
  },
  "stateless": {
    "canaryNotificationEmail": "your-email@gmail.com",
    "lambdaMemorySize": 1024,
    "randomErrorsEnabled": "false",
  },
}
`;
